### PR TITLE
Show number of connected relays in the top right corner

### LINKED ISSUE #268: Show number of connected relays in the top right

### PR DESCRIPTION
**Pull request by martindsq**
Closes #268 
It shows the number of connected relays at the top right corner of the Home Feed. The beams change to red if the number of relays is zero. I added a tap action that opens the relays list screen.

Simulator Screen Shot - iPhone 14 - 2023-06-20 at 13 40 14

### CODE REVIEW

File: Nos/Views/HomeFeedView.swift [hunk 1]
```diff
@@ -107,9 +108,32 @@ struct HomeFeedView: View {
...
+                }
+                ToolbarItem(placement: .navigationBarTrailing) {
+                    Button {
+                        router.sideMenuPath.append(SideMenu.Destination.relays)
+                        router.toggleSideMenu()
+                    } label: {
+                        HStack(spacing: 3) {
+                            Image("relay-left")
+                                .colorMultiply(numberOfConnectedRelays > 0 ? .white : .red)
+                            Text("\(numberOfConnectedRelays)")
```
Comment (mplorentz) [inline Nos/Views/HomeFeedView.swift:123]: Did you try binding this Text directly to `relayService.numberOfConnectedRelays`? RelayService is an ObservableObject so I think we should be able to do that.

**Review by mplorentz — COMMENTED** [review]
Looks pretty good. I have a few requests for changes:
- [x] The number's font is much less bold than in the designs. If you need you can use a `PlainText` and set the `.font()` and `.fontWeight()` on it.
- [x] Can we open the relay list view in a sheet? It's a little awkward to have to hit "back" and then close the side menu when we want to dismiss it.

File: Nos/Views/HomeFeedView.swift (hunk 1 above)
Comment (martindsq) [reply Nos/Views/HomeFeedView.swift]: 👌

**Comment by martindsq** [comment]
> I just noticed the text also has a shadow in the designs. We should add that too.

I'm waiting for @Chardot to send me the parameters for the shadow

**Comment by martindsq** [comment]
@mplorentz we have the green light from Sebastian to merge this without the shadow, can you re-review this PR?

**Review by mplorentz — COMMENTED** [review]
The font looks good now! I noticed a new issue which is that the "edit" mode on the relay screen is doing a little dance. Weirdly it doesn't do the same thing when opened from the Settings menu. Can you take a look at that?

https://github.com/planetary-social/nos/assets/1165004/f448ec04-e269-4470-95b4-1dd17e7b06a0

**Comment by martindsq** [comment]
@mplorentz its better now. For the record, I could not reproduce it in my iPhone 16.2 simulators, but could do it in Mac designed for iPad. The underlying cause is that the Home Feed is getting drawn every one second and the Relay list in edit mode just makes that noticeable. I fixed it by moving the sheet to the outside, I still notice a little dance at the very start but it just stops doing it so I guess we can get along with it for the moment, but we would need to take care of that redraw cycle.

**Comment by martindsq** [comment]
Its ready to be reviewed again

**Review by mplorentz — APPROVED** [review]
The jumping is now gone, and the new relay detail screen is nice! I'm clicking approve but I have one more request: I would really like to be able to copy text (like the email address or pub key) on the relay metadata view. Maybe we just need to use `SelectableText` or I think there is some other way to let people copy the text of a whole row which would be fine too.

**Comment by mplorentz** [comment]
I think I got confused because I'm asking for changes to the relay metadata screen and that's not what this PR is about. Maybe you could you throw it in a separate PR when you get a chance? Assuming it's easy.

**Pull request merged**